iib/web/api_v1.py

```diff
@@ -397,35 +397,57 @@ def get_builds() -> flask.Response:
         query = query.join(Request.user).filter(User.username == user)
 
-    if index_image:
+    if index_image or from_index:
         # https://sqlalche.me/e/20/xaj2 - Create aliases for self-join (Sqlalchemy 2.0)
         request_create_empty_index_alias = aliased(RequestCreateEmptyIndex, flat=True)
         request_add_alias = aliased(RequestAdd, flat=True)
         request_rm_alias = aliased(RequestRm, flat=True)
-        request_merge_index_image_alias = aliased(RequestMergeIndexImage, flat=True)
         request_fbc_operations_alias = aliased(RequestFbcOperations, flat=True)
 
-        query_params['index_image'] = index_image
-        # Get the image id of the image to be searched
-        image_result = Image.query.filter_by(pull_specification=index_image).first()
-        if image_result:
-            # join with the Request* tables to get the response as image_ids are stored there
-            query = (
-                query.outerjoin(
-                    request_create_empty_index_alias,
-                    Request.id == request_create_empty_index_alias.id,
-                )
-                .outerjoin(request_add_alias, Request.id == request_add_alias.id)
-                .outerjoin(
-                    request_merge_index_image_alias,
-                    Request.id == request_merge_index_image_alias.id,
-                )
-                .outerjoin(request_rm_alias, Request.id == request_rm_alias.id)
-                .outerjoin(
-                    request_fbc_operations_alias, Request.id == request_fbc_operations_alias.id
+        # join with the Request* tables to get the response as image_ids are stored there
+        query = (
+            query.outerjoin(
+                request_create_empty_index_alias,
+                Request.id == request_create_empty_index_alias.id,
+            )
+            .outerjoin(request_add_alias, Request.id == request_add_alias.id)
+            .outerjoin(request_rm_alias, Request.id == request_rm_alias.id)
+            .outerjoin(request_fbc_operations_alias, Request.id == request_fbc_operations_alias.id)
+        )
+
+        if from_index:
+            query_params['from_index'] = from_index
+            # Get the image id of the image to be searched
+            from_index_result = Image.query.filter_by(pull_specification=from_index).first()
+            if not from_index_result:
+                # if from_index is not found in image table, then raise an error
+                raise ValidationError(f'from_index {from_index} is not a valid index image')
+
+            query = query.filter(
+                or_(
+                    request_create_empty_index_alias.from_index_id == from_index_result.id,
+                    request_add_alias.from_index_id == from_index_result.id,
+                    request_rm_alias.from_index_id == from_index_result.id,
+                    request_fbc_operations_alias.from_index_id == from_index_result.id,
                 )
```

Comment:
This can be written with in_ operator., but I am unsure if it worsens or improves readability.
`query = query.filter(
                    from_index_result.id.in_(
                        [
                        request_create_empty_index_alias.from_index_id,
                        request_add_alias.from_index_id,
                        request_rm_alias.from_index_id,
                        request_fbc_operations_alias.from_index_id,
                       ]
                  )
)`

* Parameter of the in_ operator should be probably list.

Reply:
This is good idea however it is not usable for our use-case. We need to find integer in other variables. But in_ should be used to compare variable with other data. I tried your suggestion but got this error: ` AttributeError: 'int' object has no attribute 'in_'`. Therefore I would suggest to keep current query for now.

Reply:
Yes, I forgot it is intended to use with sqlalchemy object parameter.